Source/Pawnmorphs/Esoteria/Genebank/GenebankEntry.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Pawnmorph.Chambers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+
+namespace Pawnmorph.Genebank.Model
+{
+	/// <summary>
+	/// Abstract base type for a typed genebank entry.
+	/// </summary>
+	/// <typeparam name="T"></typeparam>
+	public abstract class GenebankEntry<T> : IGenebankEntry, IExposable
+	{
+		/// <summary>
+		/// The wrapped value object
+		/// </summary>
+		protected T _value;
+
+		/// <summary>
+		/// Gets or sets the wrapped object for this genebank entry.
+		/// </summary>
+		public T Value
+		{
+			get => _value;
+			set => _value = value;
```

Comment:
I'm not sure this value needs to be settable.  It might be better to make it immutable (or at least protected), especially since GenebankEntries are mostly created ad-hoc and not reused.

Reply:
This was specifically for the ienumerable canAddToDatabase function.
As opposed to create a lot of garbage by creating a new item for each value when calling that function.